src/openms_gui/source/VISUAL/Spectrum2DCanvas.cpp

```diff
@@ -1072,11 +1072,9 @@ namespace OpenMS
     {
       gradient_str = linear_gradient_.toString();
     }
-    for (Size i = 0; i < layers_.size(); ++i)
-    {
-      layers_[i].param.setValue("dot:gradient", gradient_str);
-      recalculateDotGradient_(i);
-    }
+    layers_[layers_.size()-1].param.setValue("dot:gradient", gradient_str);
```

Comment:
@lars20070 please add a check for empty layers, otherwise we have a segfault here
```suggestion
    if (layers_.empty()) return;
    layers_[layers_.size()-1].param.setValue("dot:gradient", gradient_str);
```